src/query/src/dummy_catalog.rs

```diff
@@ -177,7 +179,27 @@ impl TableProvider for DummyTableProvider {
         &self,
         filters: &[&Expr],
     ) -> datafusion::error::Result<Vec<TableProviderFilterPushDown>> {
-        Ok(vec![TableProviderFilterPushDown::Inexact; filters.len()])
+        let supported = filters
+            .iter()
+            .map(|e| {
+                // Simple filter on primary key columns are precisely evaluated.
+                if let Some(simple_filter) = SimpleFilterEvaluator::try_new(e) {
+                    if self
+                        .metadata
+                        .column_by_name(simple_filter.column_name())
+                        .and_then(|c| (c.semantic_type == SemanticType::Tag).then_some(()))
+                        .is_some()
```

Comment:
Looks like we can also use `map()` with `unwrap_or(false)`. But `then_some()` is fine.